builder.go

```diff
@@ -338,20 +339,16 @@ func (e *Expression) Literal(v any) *Expression {
 	return e
 }
 
-func Var(s string) *Expression {
-	return Expr().Var(s)
+func Var(name string) *Expression {
+	return Expr().Var(name)
 }
 
-func (e *Expression) Var(v string) *Expression {
-	if len(v) == 0 {
+func (e *Expression) Var(name string) *Expression {
+	if len(name) == 0 {
 		panic("variable name cannot be empty")
 	}
 
-	if v[0] != '$' {
-		panic(fmt.Sprintf("variable must start with $: %s", v))
-	}
-
-	e.operand = variable(v)
+	e.operand = variable(varSymbol + name)
```

Comment:
same as @janishorsts mentioned. `$` symbol is used only for serialization, so we can skip it here